firewood/src/db.rs

```diff
@@ -316,6 +319,19 @@ impl<S: ShaleStore<Node> + Send + Sync> api::DbView for DbRev<S> {
             .await
             .map_err(|e| api::Error::InternalError(Box::new(e)))
     }
+
+    #[allow(refining_impl_trait)]
+    async fn iter<K: KeyType, V>(
+        &self,
+        first_key: Option<K>,
```

Comment:
Because `KeyType` is generic, it might make this API awkward to call with `None` as you might have to specify what type of `None::<&[u8]>` even though the type has no bearing on the output. 

I would just make two methods `iter()` and `iter_from(key)`. That would also line up better with the pervasive convention of the `iter` signature not taking any arguments.

Reply:
This is a great idea